intrinsic_and_extrinsic_optimise_real_data/src/camera_calibration_real_data.cpp

```diff
@@ -0,0 +1,422 @@
+	
+
+	for (int z = 0; z != total_chessboards_detected; z++)
+	{
+		matrix_C_.push_back(cv::Mat(3, 4, CV_64FC1, &matrix_C[z][0][0]));
+		//std::cout << "matrix_C_: \r\n" << matrix_C_[z];
+	}
+
+	std::cout << "\r\n";
+
+
+	for (int z = 0; z != total_chessboards_detected; z++)
+	{
+		matrix_A[z] = matrix_B___ * matrix_C_[z] * matrix_D__;
+		// std::cout << "matrix_A: \r\n" << matrix_A[z];
+		// std::cout << "\r\n";
+	}
+
+	// Display the real and calculated matrix A here. 
+	double true_matrix_A[total_chessboards_detected][2][Number_of_internal_corners_y * Number_of_internal_corners_x];
+
+	for (int z = 0; z != total_chessboards_detected; z++)
+	{
+		for (int i = 0; i != Number_of_internal_corners_y * Number_of_internal_corners_x; i++)
+		{
+			true_matrix_A[z][0][i] = Chessboard_corners[z][i].x;
+			true_matrix_A[z][1][i] = Chessboard_corners[z][i].y;
+		}
+	}
+
+	double calculated_matrix_A[total_chessboards_detected][3][Number_of_internal_corners_y * Number_of_internal_corners_x];
+
+	for (int z = 0; z != total_chessboards_detected; z++)
+	{
+		for (int y = 0; y != 3; y++)
+		{
+			for (int x = 0; x != Number_of_internal_corners_y * Number_of_internal_corners_x; x++)
+			{
+				calculated_matrix_A[z][y][x] = matrix_A[z].at<double>(y,x);
+			}	
+		}
+	}
+
+
+	// Recreate the images but using the calculated pixel coordinates.
+	std::vector<std::vector<cv::Point2f>> Processed_Chessboard_corners;
+
+	for (int z = 0; z != total_chessboards_detected; z++)
+	{
+		std::vector<cv::Point2f> temp;
+		for (int x = 0; x != Number_of_internal_corners_y * Number_of_internal_corners_x; x++)
+		{
+			double U = calculated_matrix_A[z][0][x] / calculated_matrix_A[z][2][x]; // (s * u) / s
+			double V = calculated_matrix_A[z][1][x] / calculated_matrix_A[z][2][x]; // (s * v) / s
+			temp.push_back(cv::Point2f(U, V));
+		}	
+		Processed_Chessboard_corners.push_back(temp);
+	}
+
+	for (int z = 0; z != total_chessboards_detected; z++)
+	{
+		cv::namedWindow("Optimised Pixel Points");
+		cv::drawChessboardCorners(images[z], patternsize, Processed_Chessboard_corners[z], true);
+		cv::imshow("Optimised Pixel Points", images[z]);
+		cv::waitKey(0); 
+	}
+
+
+/*
+	// Display the 2 matrix_A: both original and calculated
+	for (int z = 0; z != total_chessboards_detected; z++)
+	{
+		std::cout << "////////////////// Image: " << z << " ////////////////// \r\n";
+		for (int i = 0; i != Number_of_internal_corners_y * Number_of_internal_corners_x; i++)
+		{
+			std::cout << "Point number: " << i << "\r\n";
+			std::cout << "Real: \r\n";
+			std::cout << "Pixel u: " << true_matrix_A[z][0][i] << "\r\n";
+			std::cout << "Pixel v: " << true_matrix_A[z][1][i] << "\r\n";
+
+			std::cout << "Calculated: \r\n";
+			std::cout << "Pixel u: " << (calculated_matrix_A[z][0][i] / calculated_matrix_A[z][2][i]) << "\r\n";
+			std::cout << "Pixel v: " << (calculated_matrix_A[z][1][i] / calculated_matrix_A[z][2][i]) << "\r\n\n";
+		}	
+	}
+*/
+
```

Comment:
It would be helpful to print some consolidated results.  For instance:
- Max pixel error
- Minimum pixel error
- root-mean-squared error